src/Drupal/DrupalExtension/Manager/DrupalAuthenticationManager.php

```diff
@@ -86,9 +86,24 @@ public function logIn(\stdClass $user)
             throw new \Exception(sprintf("No submit button at %s", $this->getSession()->getCurrentUrl()));
         }
 
+        // Store the current URL before clicking login.
+        $loginUrl = $this->getSession()->getCurrentUrl();
+
         // Log in.
         $submit->click();
 
+        // Wait for URL change after login (max 3 seconds).
+        $timeout = microtime(true) + 3;
```

Comment:
this option needs to be configurable and should be set to `0` by default to allow users to opt-in.